File: mk1/mk1/Lista3/Zad2/Szesciokat.cpp
#include "Szesciokat.hpp"

#include <cmath>

Szesciokat::Szesciokat(int bok) noexcept(true) {
    this->bok = bok;
}

double Szesciokat::Pole() noexcept(true) {
    return (3 * sqrt(3) * bok * bok) / 2;
}

double Szesciokat::Obwod() noexcept(true) {
    return 6 * bok;
}

std::string Szesciokat::Nazwa() noexcept(true) {
    return "Szesciokat";
}

File: mk1/Lista3/Zad1/Szesciokat.java
class Szesciokat extends Figura{
    public float bok;

    Szesciokat(float bok){
        this.bok = bok;
    }

    public float Pole(){
        return (float) ((float) (6*bok*bok)/(4*Math.tan(3.14159/6)));
    }

    public float Obwod(){
        return 6*bok;
    }

    public String Nazwa(){
        return "Sześciokąt";
    }
}